book/src/dev/rfcs/0008-release-planning.md

```diff
@@ -0,0 +1,190 @@
+use.  Here are two important priorities for these processes:
+
+1.  Reducing the friction of doing any individual release, allowing us to move
+    closer to a continuous deployment model;
+2.  Reducing the risk of error in the release process.
+
+These are roughly in order of priority but they're clearly related, since the
+more friction we have in the release process, the greater the risk of error,
+and the greater the risk of error, the more friction we require to prevent it.
+
+Automation helps to reduce friction and to reduce the risk of error, but
+although it helps enormously, it has declining returns after some point (in the
+sense that automating the final 5% of the work is significantly more complex
+and error-prone than automating the first 5% of the work).  So the challenge is
+to find the right "entry point" for the automated part of the system that
+strikes the right balance.
+
+One possibility is the following.  CD automation is triggered by pushing a new
+tag to the git repository.  Tags are specified as `crate-semver`, e.g.
+`zebra-network-3.2.8`, `zebrad-3.9.1`, etc.  When a new tag is pushed, the CD
+automation parses the tag to determine the crate name.  If it is `zebrad`, it
+builds a new Docker image and publishes the image and the crate.  Otherwise, it
+just publishes the crate.
+
+To publish a new version of any component crate, the process is:
+
+1.  Edit the `Cargo.toml` to increment the version number;
+2.  Update `crate/CHANGELOG.md` with a few human-readable sentences describing
+    changes since the last release (examples: [cl1], [cl2], [cl3]);
+3.  Submit a PR with these changes;
+4.  Tag the merge commit and push the tag to the git repo.
+
+[cl1]: https://github.com/ZcashFoundation/ed25519-zebra/blob/main/CHANGELOG.md
+[cl2]: https://github.com/dalek-cryptography/x25519-dalek/blob/master/CHANGELOG.md
+[cl3]: https://github.com/dalek-cryptography/curve25519-dalek/blob/master/CHANGELOG.md
+
+All subsequent steps (publishing to crates.io, building docker images, etc) are
+fully automated.
+
+Why make these choices?
+
+- Triggering on a tag, rather than some other trigger, ensures that each
+  deployment corresponds to a particular, known, state of the source
+  repository.
+
+- Editing the version number in the `Cargo.toml` should be done manually,
+  because it is the source of truth for the crate version.
+
+- Changelog entries should be written by hand, rather than auto-generated from
+  `git log`, because the information useful as part of a changelog is generally
+  not the same as the information useful as part of commit messages.  (If this
+  were not the case, changelogs would not be useful, because `git log` already
+  exists).  Writing the changelog entries by hand would be a burden if we
+  queued a massive set of changes between releases, but because releases are
+  low-friction and we control the distribution channel, we can avoid this
+  problem by releasing frequently, on a weekly or daily basis.
```

Comment:
Frequent releases also mean that everyone on the team can be confident doing releases.

This is good, because we don't have to rely on particular staff to authorise a release, do security fixes, or otherwise manage the process. (And we can effectively manage the lightweight process by consensus.)

It's difficult for release-critical staff on popular projects to take leave, and we don't want that.